packages/common/src/dbt/DbtSchemaEditor/DbtSchemaEditor.ts

```diff
@@ -57,13 +96,38 @@ export default class DbtSchemaEditor {
             | undefined;
     }
 
+    getColumnByName(modelName: string, columnName: string) {
+        const column = this.findColumnByName(modelName, columnName);
+        if (!column) {
+            throw new Error(
+                `Column ${columnName} not found in model ${modelName}`,
+            );
+        }
+        return column;
+    }
+
     hasModels() {
         const models = this.doc.get('models');
         return isSeq(models) && models.items.length > 0;
     }
 
-    // Todo: amend type once YamlModel is moved from backend to common
-    addModel(model: unknown): DbtSchemaEditor {
+    getModelColumns(modelName: string) {
+        const model = this.findModelByName(modelName);
+        if (!model) {
+            // model not found
+            return undefined;
+        }
+        const columns = model.getIn(['columns']);
+        if (!isSeq(columns)) {
+            // node is not an array
+            return undefined;
+        }
+        return columns.items
+            .filter(isMap)
+            .map<YamlColumn>((column) => column.toJSON());
```

Comment:
[non-blocker] I think it would make sense for all public methods to return `.toJSON()` value with a proper type and only private ones to use types from `yaml`

Reply:
That was my intention at the start, but I'm finding that it's not possible because the "proper" types are created from the dbt manifest, which contains much more information than what's in the schema YAML.

"Proper" types contain all the information and can be converted to YAML types.
YAML types lack information about SQL and can't be converted back to "proper" types.